### PR TITLE
FIX disable caching of 2-d filter banks

### LINKED ISSUE #175: [BUG] cache management of 2-D filter banks
A bug found by running `pyflakes` on the Kymatio source.
Originating in code by @edouardoyallon  and @gabrielhuang

It happens in a `try` statement which is itself part of an `except` statement, so the bug is really hard to trigger, but it's there.

https://github.com/kymatio/kymatio/blob/56a846f059ba5e185a7378e4a057f11bea25552a/kymatio/scattering2d/filter_bank.py#L119
`filer_file` should read as `filter_file`

I think that, long term, this caching mechanic would be more elegantly (and more stably) implemented with `joblib.Memory`, and that the calls to `print` could be replaced by `logging`.

One example of joblib-based cache manager which comes very close to what we need here is available in `librosa`, cf. https://github.com/librosa/librosa/blob/master/librosa/cache.py

### PR DESCRIPTION
**Pull request by lostanlen**
closes #175
the function filter_bank_real no longer exists
filter_bank does what filter_bank_real used to do and does not cache the filters
+ some docstring and PEP8 improvements

### CODE REVIEW

**Comment by lostanlen** [comment]
Tests are passing. Shall we squash and merge?

**Comment by janden** [comment]
I know this sounds pedantic (maybe it is, who knows?), but please separate cosmetic changes (i.e., fixing typos, breaking up long lines, etc.) from others. It makes the diffs hard to follow. Ideally, those cosmetic changes would go in a separate PR.

**Comment by lostanlen** [comment]
@janden OK, I'll remember that.

**Comment by janden** [comment]
No problem. Will refactor this one.

File: kymatio/scattering2d/filter_bank.py [hunk 1]
```diff
@@ -32,24 +32,28 @@ def filter_bank_real(M, N, J, L=8):
...
+                3.0 / 4.0 * np.pi /2**j, 4.0/L, offset=offset_unpad)
             psi_signal_fourier = fft2(psi_signal)
             for res in range(j + 1):
-                psi_signal_fourier_res = periodize_filter_fft(psi_signal_fourier, res)
-                psi[res]=torch.FloatTensor(np.stack((np.real(psi_signal_fourier_res), np.imag(psi_signal_fourier_res)), axis=2))
-                # Normalization to avoid doing it with the FFT!
+                psi_signal_fourier_res = periodize_filter_fft(
+                    psi_signal_fourier, res)
+                psi[res] = torch.FloatTensor(
+                    np.stack((np.real(psi_signal_fourier_res),
```
Comment (eickenberg) [inline kymatio/scattering2d/filter_bank.py]: @edouardoyallon re the point i was making (in a PR about plotting filters) about casting between complex and real. Here it is more serious:
This operation here is inefficient as it creates a copy. You can exploit the fact that numpy stores complex numbers as pairs of real numbers by setting a dtype and reshaping:

Check this (meaning copy and paste into an ipython):
```python
a = (np.random.randn(3, 4) + 1j* np.random.randn(3, 4)).astype('complex64')
b = a.reshape(a.shape + (1,))
b.dtype = 'float32'
print("b.shape is ", b.shape) # (3, 4, 2)
c = np.stack([np.real(a), np.imag(a)], axis=-1)
from numpy.testing import assert_array_equal
assert_array_equal(c, b)
print("a is ", a, "\n\nb is ", b, "\n\nc is ", c)
```

Don't change this in this PR. It just popped out at me as a good opportunity to explain what I was talking about a couple of days ago.

File: kymatio/scattering2d/filter_bank.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from ..caching import get_cache_dir
 
 
-def filter_bank_real(M, N, J, L=8):
+def filter_bank(M, N, J, L=8, cache=False):
```
[the remark this answers is not in the document]
Comment (lostanlen) [reply kymatio/scattering2d/filter_bank.py]: fixed in 7dedcf7

**Pull request merged**